sdk/metric/provider_test.go

```diff
@@ -174,3 +174,84 @@ func TestMeterProviderMixingOnRegisterErrors(t *testing.T) {
 		"Metrics produced for instrument collected by different MeterProvider",
 	)
 }
+
+func TestMeterProviderCardinalityLimit(t *testing.T) {
+	tests := []struct {
+		name        string
+		limit       int
+		expectedMax int // 0 => no limit
+	}{
+		{
+			name:        "no limit (default)",
+			limit:       -1, // special case: constructor without limit
+			expectedMax: 0,
+		},
+		{
+			name:        "no limit (limit=0)",
+			limit:       0,
+			expectedMax: 0,
+		},
+		{
+			name:        "no limit (negative)",
+			limit:       -5,
+			expectedMax: 0,
+		},
+		{
+			name:        "limit=5",
+			limit:       5,
+			expectedMax: 5,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := NewManualReader()
+
+			var mp *MeterProvider
+			if tt.limit == -1 {
```

Comment:
This is a hack of which the test case reader may be not aware.
You can redactor the `tests` to change `limit int` into `option metric.Option`.
An `if tt.option != nil` would seem not that strange/hacky. Also providing no option is clear that we are testing a scenario where the option is not passed.